src/common/hooks/use-user.tsx

```diff
@@ -26,10 +26,9 @@ const useFetchAccountData = (
   const apiServer = useApiServer();
 
   const fetcher = async (principal: string): Promise<AccountDataResponse> =>
-    fetchAllAccountData(apiServer)(principal);
+    fetchAllAccountData(apiServer)({ principal }) as Promise<AccountDataResponse>;
```

Comment:
Flag here - I see that if you remove this the type's don't match because `AccountDataResponse.transactions` can't be null. This is definitely a code smell that I would fix.